### PR TITLE
Added correlation ids to log messages

### PR DESCRIPTION
**Pull request by majikandy**
# Correlation id of job id on messages

Adds correlation id to the structured trace/error messages

Relates to issue #523

### Acceptance Criteria

- [ ] Correlation id of the job id against log messages where that is available

### Testing information

logs are in json and look something like this
```
{"timestamp":"2020-03-23T16:17:18.317941+00:00 ","level":"ERROR","span":{"name":"create_service_worker","state_manager":"StateManager { client: Client { connection_info: ConnectionInfo { addr: Tcp(\"127.0.0.1\", 6379), db: 0, passwd: None } } }","time_to_wait":"30s"},"target":"doc_index_updater::service_bus_client","message":"Error Couldn't retrieve file: [-31] Failed opening remote file","correlation_id":"8aacd062-8179-4852-9f9d-58d4c7f961d3"}
```
### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/service_bus_client/mod.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl DocIndexUpdaterQueue {
         T: Message,
         RetrievedMessage<T>: ProcessRetrievalError,
     {
-        tracing::info!("Checking for messages");
+        tracing::info!("Checking for messages.");
```
Comment (sgrowe) [inline medicines/doc-index-updater/src/service_bus_client/mod.rs]: Probably best if this was changed to `trace` or `debug` as it's not that informative

Comment (majikandy) [reply medicines/doc-index-updater/src/service_bus_client/mod.rs]: sounds good

File: medicines/doc-index-updater/src/delete_manager/mod.rs [hunk 2]
```diff
@@ -42,9 +42,8 @@ impl ProcessRetrievalError for RetrievedMessage<DeleteMessage> {
         e: anyhow::Error,
         state_manager: &StateManager,
     ) -> anyhow::Result<()> {
-        tracing::info!(
-            "Setting error state in state manager for job {}",
-            self.message.job_id
+        tracing::info!(message = "Setting error state in state manager",
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/delete_manager/mod.rs]: this doesn't look formatted correctly - have you run `cargo fmt` on your changes?

File: medicines/doc-index-updater/src/delete_manager/mod.rs [hunk 3]
```diff
@@ -71,45 +70,41 @@ pub async fn process_message(message: DeleteMessage) -> Result<Uuid, anyhow::Err
     let blob_name =
         get_blob_name_from_content_id(message.document_content_id.clone(), &search_client).await?;
     tracing::info!(
-        "Found blob name {} for document content ID {} from index for job {}",
-        &blob_name,
-        &message.document_content_id,
-        &message.job_id,
+        message = format!("Found blob name {} for document content ID {} from index", &blob_name, &message.document_content_id).as_str(),
+        correlation_id = &message.job_id.to_string().as_str()
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/delete_manager/mod.rs]: do you need `&` and `.as_str()`?

File: medicines/doc-index-updater/src/delete_manager/mod.rs (hunk 2 above)
Comment (majikandy) [reply medicines/doc-index-updater/src/delete_manager/mod.rs]: nope, will do

File: medicines/doc-index-updater/src/delete_manager/mod.rs (hunk 3 above)
Comment (majikandy) [reply medicines/doc-index-updater/src/delete_manager/mod.rs]: no - resolved together

File: medicines/doc-index-updater/src/search_client/mod.rs [hunk 4]
```diff
@@ -144,11 +144,12 @@ async fn search(
         .header("api-key", &config.api_key)
         .build()?;
 
-    println!("Requesting from URL: {}", &req.url());
+    tracing::debug!("Requesting from URL: {}", &req.url());
 
     client
         .execute(req)
         .await?
+        .error_for_status()?
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/search_client/mod.rs]: https://docs.rs/reqwest/0.10.0-alpha.2/reqwest/struct.Response.html#method.error_for_status

File: medicines/doc-index-updater/src/create_manager/mod.rs [hunk 5]
```diff
@@ -84,11 +93,19 @@ pub async fn process_message(message: CreateMessage) -> Result<Uuid, anyhow::Err
     let metadata: BlobMetadata = message.document.into();
     let blob = create_blob(&storage_client, &file, metadata.clone().into()).await?;
     let name = blob.name.clone();
-    tracing::info!("Uploaded blob {} for job {}", &name, &message.job_id);
-
+    
+    tracing::info!(
```
Comment (sgrowe) [inline medicines/doc-index-updater/src/create_manager/mod.rs]: I'd be tempted to make this info log & the one above both debug logs tbh as they're not particularly notable events I would say? Dunno what other people think?

File: medicines/doc-index-updater/src/create_manager/mod.rs [hunk 6]
```diff
@@ -50,8 +50,11 @@ impl ProcessRetrievalError for RetrievedMessage<CreateMessage> {
         e: anyhow::Error,
         state_manager: &StateManager,
     ) -> anyhow::Result<()> {
-        if e.to_string() == "Couldn't retrieve file: [-31] Failed opening remote file" {
-            tracing::info!("Updating state to errored and removing message");
+        if e.to_string() == "Couldn't retrieve file: [-31] Failed opening remote file".to_string() {
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/create_manager/mod.rs]: why revert this?

Comment (majikandy) [reply medicines/doc-index-updater/src/create_manager/mod.rs]: Ah, mistake, I didn’t see the subtle difference of the to_string() at the end

File: medicines/doc-index-updater/src/search_client/mod.rs (hunk 4 above)
Comment (majikandy) [reply medicines/doc-index-updater/src/search_client/mod.rs]: exactly

**Pull request merged**